### PR TITLE
Allow insecure TLS connections 

### PR DESCRIPTION
**Pull request by mancabizjak**
Since #17, client and server use TLS to secure channel over which they communicate. However, for development and testing purposes we sometimes want to avoid errors stemming from the server's use of self-signed cert. 

In this PR we:
* add an additional parameter to client's `GetConnection()`, which specifies whether to avoid checking of server's CA cert chain and hostname. 
* Update the calls in the test code and client CLI code.
* Update README with explanation and examples that demonstrate the purpose and usage of the `--insecure` flag.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -191,7 +194,27 @@ To control keys and certificates used for TLS, emmy CLI programs use several fla
...
+  2017/09/13 12:48:47 [client] 12:48:47.232 GetConnection ▶ INFO 001 Getting the connection
+  Cannot connect to gRPC server: Could not connect to server 10.12.13.45:7007 (x509: cannot validate certificate for 10.12.13.45 because it doesnt contain any IP SANs)
+  ```
+
+  Now let's include the *--insecure* flag, and the (insecure) connection to the server is now successfully established.
+
+  ```bash
+  $ emmy client --server 10.10.43.45:7007 --insecure schnorr
+
+  2017/09/13 12:49:14 [client] 12:49:14.961 GetConnection ▶ INFO 001 Getting the connection
```
Comment (miha-plesko) [inline README.md]: Perhaps a WARNING would be in place when `--insecure` is used?

File: test/communication_test.go [hunk 2]
```diff
@@ -37,7 +37,8 @@ func TestMain(m *testing.M) {
 	go server.Start(7008)
 
 	// Establish a connection to previously started server
-	testGrpcClientConn, err = client.GetConnection(testGrpcServerEndpoint, "testdata/server.pem")
+	testGrpcClientConn, err = client.GetConnection(testGrpcServerEndpoint,
+		"testdata/server.pem", true)
```
Comment (miha-plesko) [inline test/communication_test.go]: Set `insecure` to `false` here since you're passing the cert anyway.

Comment (miha-plesko) [reply test/communication_test.go]: Also, could you add another test where you don't specify the cert (use empty string), and set `insecure` to `true`?

File: cli/flags.go [hunk 3]
```diff
@@ -40,6 +40,12 @@ var caCertFlag = cli.StringFlag{
 	Usage: "`PATH` to certificate file of the CA that issued emmy server's certificate",
 }
 
+// concurrencyFlag indicates whether to run clients concurrently or not.
```
Comment (miha-plesko) [inline cli/flags.go]: Whoooops 😄

File: README.md (hunk 1 above)
Comment (mancabizjak) [reply README.md]: Indeed. Done.

File: cli/flags.go (hunk 3 above)
Comment (mancabizjak) [reply cli/flags.go]: Fixed :smile:

File: test/communication_test.go (hunk 2 above)
Comment (mancabizjak) [reply test/communication_test.go]: Done and done.

File: client/client.go [hunk 4]
```diff
@@ -22,12 +21,18 @@ func SetLogLevel(level string) error {
...
 // is able to multiplex several RPCs on the same connection, thus reducing the overhead
-func GetConnection(serverEndpoint, caCert string) (*grpc.ClientConn, error) {
+func GetConnection(serverEndpoint, caCert string, insecure bool) (*grpc.ClientConn, error) {
 	logger.Info("Getting the connection")
 	timeoutSec := config.LoadTimeout()
 
+	// Notify the end user about security implications when running in insecure mode
+	if insecure {
+		logger.Warning(`######## You requested an *insecure* channel! ######## As a consequence,
+			server's identity will *NOT* be validated! Please consider using a secure connection`)
```
Comment (miha-plesko) [inline client/client.go]: FYI: if you use backticks in Go, then it will take all the text *literally*, including  the three tabs at the beginning of your second line. Perhaps it would be nicer to have to separate warning lines, but I'm not too picky about this.
